activerecord/lib/active_record/query_logs.rb

```diff
@@ -175,12 +175,17 @@ def tag_content
           tags.flat_map { |i| [*i] }.filter_map do |tag|
             key, value_input = tag
 
-            val = case value_input
-                  when nil then tag_value(key)
-                  when Proc then instance_exec(&value_input)
-                  else value_input
+            val = if value_input.nil?
+              tag_value(key)
+            elsif value_input.respond_to?(:call)
+              if value_input.arity == 0
```

Comment:
I chose to add this arity check so that it's less error prone. I don't think it's a significant overhead (`tags.flat_map { |i| [*i] }.filter_map` is more likely to be one, and I'd like to tackle that soon).